src/types/types.ts

```diff
@@ -297,7 +297,9 @@ export type Stats = {
  */
 
 export type Key = {
+  uid: string
   description: string
+  name: string | null
```

Comment:
Why can the name be `null` but not the uid?

Reply:
Ok but below you have 
```js
export type KeyCreation = {
  uid?: string
  name?: string
```

With `uid` optional, which confuses me

Reply:
Key is the returned information when you get a Key. While KeyCreation are the parameters when you create a key! 

As per this part of the sentence
> Add the possibility to specify :uid to generate deterministic API keys **if null Meilisearch generates one**.


But! The returned key will always return with an `uid`. The `uid` will either be the one you chose or a generated one by Meilisearch! 

```js
client.createKey(options: KeyCreation): Promise<Key>
```

Create key expect the following parameters (as per [the spec](https://github.com/meilisearch/specifications/blob/d637c11c6507cea207ed299692d501593a9064e8/text/0085-api-keys.md#3244-post---keys))
```ts
export type KeyCreation = {
  uid?: string
  name?: string
  description?: string
  actions: string[]
  indexes: string[]
  expiresAt: string | null
}
```

When created you will receive a `Key` object with the following fields (as per [the spec](https://github.com/meilisearch/specifications/blob/d637c11c6507cea207ed299692d501593a9064e8/text/0085-api-keys.md#3241-api-key-resource-representation)): 

```js
export type Key = {
  uid: string
  description: string
  name: string | null
  key: string
  actions: string[]
  indexes: string[]
  expiresAt: string
  createdAt: string
  updateAt: string
}
```